Code/src/main/java/nl/utwente/viskell/haskell/env/CatalogFunction.java

```diff
@@ -33,6 +37,31 @@
         this.isCommon = isCommon;
     }
 
+    @Override
+    public Map<String, Object> toBundleFragment() {
+        Map<String, Object> bundleFragment = new HashMap<>();
+        bundleFragment.put(Bundleable.KIND, this.getClass().getSimpleName());
+        bundleFragment.put("name", name);
+        bundleFragment.put("signature", signature.toString());
+        bundleFragment.put("category", category);
+        bundleFragment.put("documentation", documentation);
+        bundleFragment.put("isConstructor", isConstructor);
+        bundleFragment.put("isCommon", isCommon);
```

Comment:
I did't know enough about how all this works to for that idea to even occur to me.
If it can be done, then great and would simplify this serialization a lot.

Any example code to lookup a CatalogFunction from the name and instantiate it, would be appreciated.